main.py

```diff
@@ -28,25 +30,26 @@ def main(config: Annotated[Path, typer.Option(help="Path to test config file")])
         # Check if the required keys are present in the test case
         result, missing_keys = check_keys_in_list(test_case['test'], required_keys)
         if result:
-            # Process the data if the required keys are present
-            source_df, target_df = process_data(test_case)
-
-            # Convert the dataframes to dictionaries for comparison
-            source_data = source_df.to_dict(orient='records')
-            target_data = target_df.to_dict(orient='records')
-            pre_process = test_case['test']['preprocess']
 
             # Performing preprocessing if available
-            sharepoint_file = pre_process.get("sharepoint_download")
+            pre_processor = test_case['test']['preprocessor']
+            sharepoint_file = pre_processor.get("sharepoint_download")
```

Comment:
This might be better served to pass it to the preprocessor as sharepoint may not always be defined in the test case.

Reply:
Implemented iteration